build android template under windows 7 32bit fails

**OS : windows 7 32bit**
**Cannot build android template under windows 7 32bit**
uses [scipt by @Calinou](https://gist.github.com/Calinou/44edbd5d9c4a2ac0dbc8f165b8b7f0ce) (with modified ENVVARS)
```
C:\Windows\system32>cd c:\godot

c:\godot>set threads=1
c:\godot>set PATH=C:\mingw\mingw32\bin
c:\godot>set JAVA_HOME=C:\Program Files\java\jre1.8.0_111
c:\godot>set ANDROID_HOME=C:\Program Files\Android\android-sdk
c:\godot>set ANDROID_NDK_ROOT=C:\android-ndk-r13b
c:\godot>set NDK_TARGET=arm-linux-androideabi-4.9

c:\godot>call "C:\Python27\Scripts\scons.bat" p=android -j1 verbose=yes tools=no
 target=release_debug
scons: Reading SConscript files ...

MSVC not detected, attempting Mingw.
- gcc detected.
Godot Android!!!!! (armv7) (with neon)
scons: done reading SConscript files.
scons: Building targets ...
C:\android-ndk-r13b/toolchains/llvm/prebuilt/windows-x86/bin/clang++ -o platform
\android\os_android.os -c -fno-rtti -fno-exceptions -DNO_SAFE_CAST -DFREETYPE_EN
ABLED -DENABLE_DEPRECATED -isystem C:\android-ndk-r13b/platforms/android-14/arch
-arm/usr/include -Wno-invalid-command-line-argument -Wno-unused-command-line-arg
ument -fpic -ffunction-sections -funwind-tables -fstack-protector-strong -fvisib
ility=hidden -fno-strict-aliasing -Wa,--noexecstack -DANDROID -DNO_STATVFS -DGLE
S2_ENABLED -D__ARM_ARCH_7__ -D__ARM_ARCH_7A__ -march=armv7-a -mfloat-abi=softfp
-mfpu=neon -D__ARM_NEON__ -target armv7-none-linux-androideabi -fno-integrated-a
s -gcc-toolchain C:\android-ndk-r13b/toolchains/arm-linux-androideabi-4.9/prebui
lt/windows-x86 -O2 -DNDEBUG -ffast-math -funsafe-math-optimizations -fomit-frame
-pointer -ftree-vectorize -DDEBUG_ENABLED -DANDROID_ENABLED -DUNIX_ENABLED -DNO_
FCNTL -DMPC_FIXED_POINT -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -Icor
e -Icore\math -Itools -Idrivers -I. -Iplatform\android -Ithirdparty\zlib -Ithird
party\freetype -Ithirdparty\freetype\include -Ithirdparty\libpng platform\androi
d\os_android.cpp
=====
In file included from platform\android\os_android.cpp:29:
In file included from platform\android/os_android.h:32:
In file included from core\os/input.h:32:
core\object.h:522:8: warning: 'this' pointer cannot be null in well-defined C++
code; pointer may be assumed to always convert to true [-Wundefined-bool-convers
ion]
                if (!this)
                    ~^~~~
core\object.h:537:8: warning: 'this' pointer cannot be null in well-defined C++
code; pointer may be assumed to always convert to true [-Wundefined-bool-convers
ion]
                if (!this)
                    ~^~~~
2 warnings generated.
C:\mingw\mingw32\bin\as.exe: unknown option -- mfpu=neon
clang++.exe: error: assembler command failed with exit code 1 (use -v to see inv
ocation)

=====
scons: *** [platform\android\os_android.os] Error 1
scons: building terminated because of errors.
```

--- a/platform/android/detect.py
+++ b/platform/android/detect.py
@@ -130,18 +130,17 @@ def configure(env):
         else:
             env.extra_suffix = ".armv7" + env.extra_suffix
 
+    mt_link = True
     if (sys.platform.startswith("linux")):
-        if (platform.machine().endswith('64')):
-            host_subpath = "linux-x86_64"
-        else:
-            host_subpath = "linux-x86"
+        host_subpath = "linux-x86_64"
     elif (sys.platform.startswith("darwin")):
         host_subpath = "darwin-x86_64"
     elif (sys.platform.startswith('win')):
         if (platform.machine().endswith('64')):
             host_subpath = "windows-x86_64"
         else:
-            host_subpath = "windows-x86"
+            mt_link = False
+            host_subpath = "windows"
 
     compiler_path = env["ANDROID_NDK_ROOT"] + \
         "/toolchains/llvm/prebuilt/" + host_subpath + "/bin"
@@ -205,14 +204,15 @@ def configure(env):
         env['SHLIBSUFFIX'] = '.so'
 
     env['LINKFLAGS'] = ['-shared', '--sysroot=' +
-                        sysroot, '-Wl,--warn-shared-textrel',
-                        '-Wl,--threads']
+                        sysroot, '-Wl,--warn-shared-textrel']
     env.Append(LINKFLAGS=string.split(
         '-Wl,--fix-cortex-a8'))
     env.Append(LINKFLAGS=string.split(
         '-Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now'))
     env.Append(LINKFLAGS=string.split(
         '-Wl,-soname,libgodot_android.so -Wl,--gc-sections'))
+    if mt_link:
+        env.Append(LINKFLAGS=['-Wl,--threads'])
     env.Append(LINKFLAGS=target_opts)
     env.Append(LINKFLAGS=common_opts)
 

